src/components/NFTPage/Recommendation.vue

```diff
@@ -88,60 +94,139 @@ import { Swiper, SwiperSlide } from 'vue-awesome-swiper';
 import { DEFAULT_RECOMMENDATIONS_LIST } from '~/constant';
 
 import bookstoreMixin from '~/mixins/bookstore';
+import nftOrCollection from '~/mixins/nft-or-collection';
+import walletMixin from '~/mixins/wallet';
+import { mapActions, mapGetters } from 'vuex';
+
+const DISPLAY_ITEM_COUNT = 5;
 
 export default {
   name: 'NFTPageRecommendation',
   components: {
     Swiper,
     SwiperSlide,
   },
-  mixins: [bookstoreMixin],
+  mixins: [bookstoreMixin, nftOrCollection, walletMixin],
   props: {
-    iscnOwner: {
+    classId: {
+      type: String,
+      default: '',
+    },
+    collectionId: {
       type: String,
-      default: undefined,
+      default: '',
     },
-    isFollowed: {
+    // if true, show author works, else show default featured works
+    isAuthorSpecific: {
       type: Boolean,
-      default: false,
+      default: true,
     },
-    recommendedList: {
-      type: Array,
-      default: () => [],
+  },
+  computed: {
+    ...mapGetters(['walletHasLoggedIn', 'walletFollowees']),
+    type() {
+      if (this.nftIsWritingNFT) {
+        return 'writingNFT';
+      }
+      if (this.classId) {
+        return 'book';
+      }
+      return 'collection';
     },
-    isLoading: {
-      type: Boolean,
-      default: false,
+    isFollowed() {
+      return this.walletFollowees?.includes(this.productOwner) || false;
     },
-    shouldShowFollowButton: {
-      type: Boolean,
-      default: false,
+    shouldShowFollowButton() {
+      return (
+        this.wallet !== this.getAddress &&
+        this.walletHasLoggedIn &&
+        !this.isFollowed
+      );
     },
-    isBookNft: {
-      type: Boolean,
-      default: false,
+    hasPublisher() {
+      if (this.type === 'collection') {
+        return true;
+      }
+      return !!this.iscnWorkAuthor;
     },
-    displayItemCount: {
-      type: [Number, String],
-      default: 5,
+    defaultFeaturedWNFT() {
+      return DEFAULT_RECOMMENDATIONS_LIST.WNFT.map(nft => ({
+        classId: nft,
+      }));
     },
-    shouldShowIscnOwner: {
-      type: Boolean,
-      default: true,
+    defaultFeaturedBooks() {
+      return this.bookstoreListItemForLandingPage?.length
+        ? this.bookstoreListItemForLandingPage
+        : DEFAULT_RECOMMENDATIONS_LIST.BOOK.map(nft => ({ classId: nft }));
     },
-    iscnWorkAuthor: {
-      type: String,
-      default: undefined,
+    defaultFeaturedList() {
+      if (this.nftIsWritingNFT) {
+        return this.defaultFeaturedWNFT;
+      }
```

Comment:
Maybe we should use `this.type` here

Reply:
[🎨 Use formatted nftCollection from mixin](https://github.com/likecoin/liker-land/pull/1863/commits/2b4fd599226d95c07320b6200c4cb3723d922359)